AssemblerLib/MeshComponentMap.h

```diff
@@ -107,8 +107,29 @@ class MeshComponentMap
     /// | l_m      | comp_id_2   | gi78        |
     /// | ...      |  ...        | ...         |
     /// | l_n      | comp_id_n   | gi89        |
+    template <typename T_INT_TYPE>
+    std::vector<T_INT_TYPE> getGlobalIndicesByLocation(const std::vector<Location> &ls) const;
+
+    /// The same function as getGlobalIndicesByLocation but getting indices by component
+    template <typename T_INT_TYPE>
+    std::vector<T_INT_TYPE> getGlobalIndicesByComponent(const std::vector<Location> &ls) const;
+
+    /// Get the flags of the ghost location at location \c l.
+    ///
+    /// | Location | ComponentID | GlobalIndex |  Ghost flag |
+    /// | -------- | ----------- | ----------- |-------------|
+    /// | l        | c           | gi          |is_ghost     |
+    std::vector<bool> getGhostFlags(const Location &l) const;
+
+    /// Get the number of global unknowns.
+    std::size_t getNGlobalUnknowns() const
+    {
+        return _num_global_dof;
+    }
+
+    /// Get the ghost flags of the nodes at location \c l.
     template <ComponentOrder ORDER>
-    std::vector<std::size_t> getGlobalIndices(const std::vector<Location> &ls) const;
```

Comment:
@norihiro-w  This function is split into two functions:
```
 template <typename T_INT_TYPE>
 std::vector<T_INT_TYPE> getGlobalIndicesByLocation(const std::vector<Location> &ls) const;

/// The same function as getGlobalIndicesByLocation but getting indices by component
template <typename T_INT_TYPE>
std::vector<T_INT_TYPE> getGlobalIndicesByComponent(const std::vector<Location> &ls) const;
```
The definitions of these two functions are given in a new added file:
``` 
AssemblerLib/MeshComponentMap_imp.h.
```